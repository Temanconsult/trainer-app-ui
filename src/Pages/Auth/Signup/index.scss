.signup-wrapper {
  display: flex;
  flex-direction: row;

  .left-section {
    height: 100%;
    width: 50%;
    display: none;
    background: linear-gradient(124deg, #ede4ff 50%, #e6dbfb 50%);

    .image {
      height: 100%;
      width: 100%;
      display: flex;
      justify-content: center;
      align-items: center;
    }
  }

  .right-section {
    display: flex;
    flex-direction: column;
    justify-content: center;
    position: relative;
    width: 100%;
    text-align: center;

    .content {
      display: flex;
      flex-direction: column;
      align-items: center;
      
      .logo-wrapper {
        margin-bottom: 1.5rem;
      }
  
      .welcome {
        text-align: center;
        margin-bottom: 1.5rem;
        color: black;
        opacity: 0.5;
        font-size: 1em;
      }
  
      form {
        width: 90%;
  
        .input-wrapper,
        .full {
          display: flex;
          flex-direction: column;
          padding: 12px 0;
          width: 100%;
          animation: fadeIn .7s ease-in-out;
      
          .input,
          .input-error {
            margin: 0;
            padding: 0.5rem 0;
            border: none;
            border-bottom: 1px solid #d0d4d0;
            outline: none;
            font-size: .75rem;
            width: 100%;
            transition: border-bottom .4s ease-in, transform .4s ease-in;
          }

          .input:focus {
            border-bottom: 1px solid #a77aff;
            transform: scale(1.02);
          }
    
          .input-error {
            border-bottom: 1px solid red;
          }
        }   
          
        .full {
          width: 100%;
        }
  
        .terms-and-condition {
          display: flex;
          align-items: center;
          margin-top: 30px;
  
          span {
            margin-left: 0.5rem;
            font-style: italic;
            font-style: 1em;
            opacity: 0.75;
          }
        }

        .btn-container {
          margin-top: 2.5rem;
        }
    
          .submit,
          .google,
          .facebook {
            color: #fff;
            background-color: #A77BFF;
            width: 100%;
            margin: 0.5rem;
            padding: 0.75rem 1rem;
            font-size: 1em;
            outline: none;
            text-align: center;
            border-radius: 9999px;
          }

          .submit,
          .facebook {
            border: none;
          }
    
          .google {
            background-color: #FFF;
            border: 1px solid #1F98E9;
            color: #1F98E9;
          }
    
          .facebook {
            background-color: #5E93F7;
            color: #FFF;
          }

          .login-redirect {
            padding: 20px 0;

            .login-link {
              text-align: center;
              color: black;
              font-size: 1em;
              text-decoration: underline;
      
              .sign-in {
                color: #5E93F7;
              }
            }
          }
      }

      @media(min-width: 768px) {
        form {
          width: 60%;
        }
      }

      @media(min-width: 1024px) {
        form {
          width: 70%;
        }
      }
    }

    .terms-wrapper {
      text-align: left;
    }

    .terms {
      font-size: 0.75;
      color: gray;
    }

    .error {
      font-size: 12px;
      color: red;
      margin-top: 5px;
      animation: fadeIn .75s ease-in-out;
      text-align: left;
    };

    @media (min-width: 768px) {
      .tc {
        position: absolute;
        bottom: 5px;
        width: 100%;
      }
    }
  }

  @media(min-width: 1024px) {
    .left-section {
      display: block;
    }
    .right-section {
      width: 50%;
    }
  }

  @keyframes fadeIn {
    from { opacity: 0}
    to { opacity: 1}
  }
}

@media(min-width: 768px) {
  .signup-wrapper {
    height: 100vh;;
  }
}