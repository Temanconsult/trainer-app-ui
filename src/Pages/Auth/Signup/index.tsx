import React from 'react';
import {Link} from 'react-router-dom';
import {ReactComponent as SignupImage} from '../../../assets/images/login.svg';
import {ReactComponent as Logo} from '../../../assets/images/logo.svg';

const Signup = () => {
  return (
    <div className="h-screen flex flex-row">
      <div
        className="h-full w-1/2 hidden lg:block"
        style={{
          backgroundImage: 'linear-gradient(124deg, #ede4ff 50%, #e6dbfb 50%)',
        }}
      >
        <div className="h-full w-full p-5 flex justify-center items-center">
          <SignupImage />
        </div>
      </div>
      <div className="flex flex-col justify-center relative w-full lg:w-1/2 ">
        <div className="flex justify-center mb-4">
          <Logo />
        </div>
        <span className="text-center mb-10 text-black opacity-50">
          Please complete to create your account
        </span>
        <div className="flex flex-col justify-center items-center w-full">
          <form className="flex flex-col w-10/12 md:w-8/12 h-full">
            <div className="flex justify-between flex-col md:flex-row lg:flex-row">
              <input
                type="text"
                placeholder="First name"
                className="my-4 py-2 border-gray-400 border-b text-xs outline-none w-full md:mr-2 lg:mr-2 md:w-1/2 lg:w-1/2"
              />
              <input
                type="text"
                placeholder="Last name"
                className="my-4 py-2 border-gray-400 border-b text-xs outline-none w-full md:ml-2 lg:ml-2 md:w-1/2 lg:w-1/2"
              />
            </div>
            <input
              type="text"
              placeholder="Username"
              className="my-4 py-2 border-gray-400 border-b text-xs outline-none"
            />
            <input
              type="email"
              placeholder="Email"
              className="my-4 py-2 border-gray-400 border-b text-xs outline-none"
            />
            <input
              type="password"
              placeholder="Password"
              className="my-4 py-2 border-gray-400 border-b text-xs outline-none"
            />
            <input
              type="password"
              placeholder="Confirm Password"
              className="my-4 py-2 border-gray-400 border-b text-xs outline-none"
            />
            <div className="flex justify-between mt-6 items-center">
              <div className="flex items-center">
                <input type="checkbox" />
                <span className="ml-2 italic text-xs opacity-75">
                  I agree with terms and conditionss
                </span>
              </div>
            </div>
            <div className="flex justify-center flex-col mt-10 items-center">
              <button
                type="submit"
                className="btn bg-custom-primary btn-bg text-white w-full md:w-1/2 lg:w-full xl:w-1/2"
              >
                Sign up
              </button>
              <Link
                to="#"
                className="btn border border-custom-twitter text-custom-twitter w-full md:w-1/2 lg:w-full xl:w-1/2 mt-2"
              >
                Sign up with Google
              </Link>
              <Link to="#" className="btn bg-custom-fb w-1/2 mt-2 text-white w-full md:w-1/2 lg:w-full xl:w-1/2">
                Sign up with Facebook
              </Link>

              <Link
                to="/login"
                className="text-center mt-10 text-black text-base underline"
              >
                <span>
                  Already have an account?&nbsp;
                  <span className="text-custom-fb">Sign in.</span>
                </span>
              </Link>
            </div>
          </form>
        </div>
        <span className="absolute left-0 right-0 bottom-0 p-4 text-xs text-center text-custom-gray">
          Term of use. Privacy policy
        </span>
      </div>
    </div>
  );
};

export default Signup;
