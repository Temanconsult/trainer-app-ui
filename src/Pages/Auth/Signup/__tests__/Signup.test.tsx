import React from 'react';
import {render} from '@testing-library/react';
import {BrowserRouter as Router} from 'react-router-dom';
import Signup from '..';

describe('<Signup />', () => {
  test('renders signup component elements correctly', () => {
    const {container} = render(
      <Router>
        <Signup />
      </Router>
    );
    expect(container).toMatchSnapshot();
  });
});
