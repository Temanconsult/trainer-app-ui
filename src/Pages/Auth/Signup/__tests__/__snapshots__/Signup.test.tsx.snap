// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Signup /> renders signup component elements correctly 1`] = `
<div>
  <div
    class="signup-wrapper"
  >
    <div
      class="left-section"
    >
      <div
        class="image"
      >
        <svg>
          login.svg
        </svg>
      </div>
    </div>
    <div
      class="right-section"
    >
      <div
        class="content"
      >
        <div
          class="logo-wrapper"
        >
          <svg>
            logo.svg
          </svg>
        </div>
        <span
          class="welcome"
        >
          Please complete to create your account
        </span>
        <form>
          <div
            class="form-wrapper-half"
          >
            <div
              class="input-wrapper"
            >
              <input
                class="input"
                name="firstname"
                placeholder="First name"
                type="text"
              />
            </div>
            <div
              class="input-wrapper"
            >
              <input
                class="input"
                name="lastname"
                placeholder="Last name"
                type="text"
              />
            </div>
          </div>
          <div
            class="input-wrapper full"
          >
            <input
              class="input"
              name="username"
              placeholder="Username"
              type="text"
            />
          </div>
          <div
            class="input-wrapper full"
          >
            <input
              class="input"
              name="email"
              placeholder="Email"
              type="text"
            />
          </div>
          <div
            class="input-wrapper full"
          >
            <input
              class="input"
              name="password"
              placeholder="Password"
              type="password"
            />
          </div>
          <div
            class="input-wrapper full"
          >
            <input
              class="input"
              name="confirmPassword"
              placeholder="Confirm Password"
              type="password"
            />
          </div>
          <div
            class="terms-wrapper"
          >
            <div
              class="terms-and-condition"
            >
              <input
                name="termsAndCondition"
                type="checkbox"
              />
              <span>
                I agree with terms and conditionss
              </span>
            </div>
          </div>
          <div
            class="btn-container"
          >
            <button
              class="submit"
              type="submit"
            >
              Sign up
            </button>
            <button
              class="google"
              type="button"
            >
              Sign up with Google
            </button>
            <button
              class="facebook"
              type="button"
            >
              Sign up with Facebook
            </button>
            <div
              class="login-redirect"
            >
              <a
                class="login-link"
                href="/login"
              >
                <span>
                  Already have an account?Â 
                  <span
                    class="sign-in"
                  >
                    Sign in.
                  </span>
                </span>
              </a>
            </div>
          </div>
        </form>
      </div>
      <div
        class="tc"
      >
        <span
          class="terms"
        >
          Term of use. Privacy policy
        </span>
      </div>
    </div>
  </div>
</div>
`;
